plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -232,19 +233,26 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
-    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType)
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
     {
-        if (fromType.isShort()) {
-            return new ShortDecimalToVarcharCoercer(fromType);
+        if (toType.isUnbounded()) {
```

Comment:
It doesn't have to be as verbose.

String values are limited in length to int anyway, so the unbounded varchar is just a simplification.
```java
int lengthBound = toType.getLength().orElse(Integer.MAX_VALUE);
```

and remove the class duplication.